app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/messages/IEMessage.scala

```diff
@@ -35,4 +37,13 @@ trait IEMessage {
   def toXml: NodeSeq
 
   def lrnEquals(lrn: String): Boolean
+}
+
+object IEMessage {
```

Comment:
I would have this as a trait and make the IE837Message and the IE871Message derived from this. Putting this as an object ,as companion of the IEMessage suggest all the messages should have this, while with the trait we make a distinct separation:

`
trait SubmitterTypeConverter {
  def convert(submitterType: SubmitterType): ExciseTraderType = {
    submitterType match {
      case Number1Value31 => Consignor
      case Number2Value30 => Consignee
    }
  }
}


case class IE837Message
(
  private val obj: IE837Type,
  private val key: Option[String],
  private val namespace: Option[String]
) **extends IEMessage with SubmitterTypeConverter** {

  def submitter: ExciseTraderType = **convert**(obj.Body.ExplanationOnDelayForDelivery.AttributesValue.SubmitterType)
  def consignorId: Option[String] = submitter match {
    case Consignor => Some(obj.Body.ExplanationOnDelayForDelivery.AttributesValue.SubmitterIdentification)
    case _ => None
  }
.....
}

`

Reply:
done